compiler/src/dotty/tools/dotc/typer/Applications.scala

```diff
@@ -1977,8 +1977,14 @@ trait Applications extends Compatibility {
            *   new java.io.ObjectOutputStream(f)
            */
           pt match {
-            case SAMType(mtp) => narrowByTypes(alts, mtp.paramInfos, mtp.resultType)
-            case _ => compat
+            case SAMType(mtp) =>
+              narrowByTypes(alts, mtp.paramInfos, mtp.resultType)
+            case _ =>
+              // pick any alternatives that are not methods since these might be convertible
+              // to the expected type, or be used as extension method arguments.
+              val convertible = alts.filterNot(alt =>
+                  normalize(alt, IgnoredProto(pt)).widenSingleton.isInstanceOf[MethodType])
```

Comment:
Is using `normalize` really needed here or could we do:
```suggestion
                  alt.widenSingleton.isInstanceOf[MethodOrPoly])
```

Reply:
It's needed for skipping using clauses

Reply:
I see. `normalize` will create fresh type variables for PolyTypes, they'll be gc'ed eventually but if we wanted to avoid creating them in the first place we could perhaps have a `stripPolyAndImplicit` method, or add a flag to `normalize` for situations where we don't keep around its result.